apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -503,24 +506,41 @@ func (dc *CassandraDatacenter) GetClusterLabels() map[string]string {
 	}
 }
 
+// apimachinery validation does not expose this, copied here
+const dns1035LabelFmt string = "[a-z]([-a-z0-9]*[a-z0-9])?"
+
+func cleanupForKubernetes(input string) string {
```

Comment:
Would it make more sense to restrict the C* cluster name to a valid dns name? I just wonder if sanitizing the service names in this way make us more susceptible to naming conflicts.

Reply:
@burmanm astutely pointed out to me that restricting the cluster name to a vaild dns name would present problems with migrating C* clusters that are running outside of k8s. Given this, I think the changes make sense as is.